op-chain-ops/crossdomain/precheck.go

```diff
@@ -67,7 +67,13 @@ func PreCheckWithdrawals(db *state.StateDB, withdrawals []*LegacyWithdrawal) ([]
 			continue
 		}
 
-		filtered = append(filtered, slotsInp[slot])
+		wd := slotsInp[slot]
+		if wd.Sender == nil || *wd.Sender == predeploys.L2CrossDomainMessengerAddr {
```

Comment:
Should this be `wd.Sender != predeploys.L2CrossDomainMessengerAddr`?

Reply:
Yes. Nice catch. Will unit test these methods to make sure I don't introduce any further issues here.